File: src/main/java/io/Cbuild/gnu_make_functions/compiler/I_compiler_base.java
package io.Cbuild.gnu_make_functions.compiler;

import io.Cbuild.cBuildIR;
import io.Cbuild.cbuildParser;
import io.Cbuild.gnu_make_functions.make_function_dispatcher;
import io.Cbuild.gnu_make_functions.type;

public interface I_compiler_base {

     cBuildIR.FunctionIR compile(cbuildParser.FunctionContext ctx, type.MakeFunction function);
}

File: src/gnu-make-functions/compiler/Icompiler-base.ts
import { FunctionContext } from "@parser/cbuildParser";
import { FunctionIR } from "@compiler/ir";
import type { MakeFunction } from "@gnu-make-functions/type";

export interface I_compiler_base {
  compile(ctx: FunctionContext, func: MakeFunction): FunctionIR;
}
